### PR TITLE
Split effect headers and a bit of cleanup

### PR DESCRIPTION
**Pull request by AngheloAlf**
Moves functions out of `functions.h` and removed most of the extern variables from `variables.h`.

I made an individual header of each one of the `z_eff_*.c` files.
I made a few defines for flags and used them everywhere I found and fixed a few cases where a raw value was being passed to `Effect_Add` instead of the enum.

### CODE REVIEW

File: include/z64effect_ss.h [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+// void EffectSsBlast_SpawnShockwave(UNK_TYPE1 param_1, UNK_TYPE1 param_2, UNK_TYPE1 param_3, UNK_TYPE1 param_4, UNK_TYPE4 param_5, UNK_TYPE4 param_6, UNK_TYPE2 param_7);
+void EffectSsBlast_SpawnWhiteShockwave(struct PlayState* play, Vec3f* pos, Vec3f* velocity, Vec3f* accel);
+// void EffectSsGSpk_SpawnAccel(UNK_TYPE4 uParm1, UNK_TYPE4 uParm2, Vec3f* pzParm3, Vec3f* pzParm4, Vec3f* param_5, Color_RGBA8* param_6, Color_RGBA8* param_7, UNK_TYPE2 param_8, UNK_TYPE2 param_9);
+// void EffectSsGSpk_SpawnNoAccel(UNK_TYPE1 param_1, UNK_TYPE1 param_2, UNK_TYPE1 param_3, UNK_TYPE1 param_4, UNK_TYPE4 param_5, UNK_TYPE4 param_6, UNK_TYPE4 param_7, UNK_TYPE2 param_8, UNK_TYPE2 param_9);
+void EffectSsGSpk_SpawnFuse(struct PlayState* play, struct Actor* actor, Vec3f* pos, Vec3f* velocity, Vec3f* accel);
+// void EffectSsGSpk_SpawnRandColor(UNK_TYPE1 param_1, UNK_TYPE1 param_2, UNK_TYPE1 param_3, UNK_TYPE1 param_4, UNK_TYPE4 param_5, UNK_TYPE2 param_6, UNK_TYPE2 param_7);
+// void EffectSsGSpk_SpawnSmall(UNK_TYPE1 param_1, UNK_TYPE1 param_2, UNK_TYPE1 param_3, UNK_TYPE1 param_4, UNK_TYPE4 param_5, UNK_TYPE4 param_6, UNK_TYPE4 param_7);
```
Comment (engineer124) [inline include/z64effect_ss.h:132]: Would you mind cleaning up these functions here? Either uncomment them or remove them if they should be exposed or not

File: src/code/z_eff_blure.c [hunk 2]
```diff
@@ -726,10 +744,9 @@ void EffectBlure_SetupSimpleAlt(GraphicsContext* gfxCtx, EffectBlure* this, Vtx*
 void (*sSetupHandlers[])(GraphicsContext* gfxCtx, EffectBlure* this, Vtx* vtx) = {
     EffectBlure_SetupSimple,
     EffectBlure_SetupSimpleAlt,
+    NULL,
 };
```
Comment (engineer124) [inline src/code/z_eff_blure.c:748]: What would happen if you try to run a function with `NULL`? There's no safety checks to see if `drawMode` is less than 2, are you sure `NULL` should be in here?

Comment (hensldm) [reply src/code/z_eff_blure.c:748]: Agreed here. Probably better to keep the unused pad data.

Comment (AngheloAlf) [reply src/code/z_eff_blure.c:748]: `sSetupHandlers` is only used on `EffectBlure_DrawSimple`, which has a check for `drawMode` being less than 2
https://github.com/zeldaret/mm/blob/b2535fc75a845adffbec760f58c2d654b99f2964/src/code/z_eff_blure.c#L1022-L1026

File: src/code/z_effect_soft_sprite_old_init.c [hunk 3]
```diff
@@ -175,9 +178,10 @@ void func_800B139C(PlayState* play, Vec3f* pos, Vec3f* velocity, Vec3f* accel, C
 
 void func_800B13D8(Vec3f* srcPos, f32 randScale, Vec3f* newPos, Vec3f* velocity, Vec3f* accel) {
     s16 randAngle;
-    f32 rand = Rand_ZeroOne() * randScale;
+    f32 rand;
 
-    randAngle = (Rand_ZeroOne() * 65536.0f);
+    rand = Rand_ZeroOne() * randScale;
```
[the remark this answers is not in the document]
Comment (AngheloAlf) [reply src/code/z_effect_soft_sprite_old_init.c:183]: I was trying to get merge the declaration and definition but I wasn't able to. Then I forgot to undo what I did, whoops

File: src/code/z_eff_blure.c (hunk 2 above)
Comment (hensldm) [reply src/code/z_eff_blure.c]: Hmm, with the check it seems fine to actually move the NULL inside. I'm wondering if we should add a drawMode enum though?

File: include/z64eff_blure.h [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    /* 0x18C */ s16 addAngle;
+    /* 0x18E */ u8 p1StartColor[EFFECT_BLURE_COLOR_COUNT];
+    /* 0x192 */ u8 p2StartColor[EFFECT_BLURE_COLOR_COUNT];
+    /* 0x196 */ u8 p1EndColor[EFFECT_BLURE_COLOR_COUNT];
+    /* 0x19A */ u8 p2EndColor[EFFECT_BLURE_COLOR_COUNT];
+    /* 0x19E */ u8 numElements; // "now_edge_num"
+    /* 0x19F */ u8 elemDuration;
+    /* 0x1A0 */ u8 unkFlag;
+    /* 0x1A1 */ u8 drawMode; // EffectBlureDrawMode enum
+    /* 0x1A2 */ Color_RGBA8 altPrimColor; // used with drawMode 1
```
Comment (hensldm) [inline include/z64eff_blure.h:83]: Same here

**Review by hensldm — APPROVED** [review]
Just a comment about some comments.

File: include/z64eff_blure.h [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+struct GraphicsContext;
+
+#define EFFECT_BLURE_COLOR_COUNT 4
+
+//#define EFFECT_BLURE_FLAG_1         (1 << 0)
+#define EFFECT_BLURE_FLAG_2         (1 << 1)
+#define EFFECT_BLURE_FLAG_4         (1 << 2)
+//#define EFFECT_BLURE_FLAG_8         (1 << 3)
+#define EFFECT_BLURE_FLAG_10        (1 << 4)
```
Comment (engineer124) [inline include/z64eff_blure.h:17]: Is there a reason flags 1 and 8 are commented out?

Comment (AngheloAlf) [reply include/z64eff_blure.h:17]: Because they are not used and I forgot to remove them, whoops :sweat_smile:

**Review by engineer124 — APPROVED** [review]

**Pull request merged**